src/scripts/script-update-block.ts

```diff
@@ -0,0 +1,79 @@
+import 'dotenv/config';
+
+import { Connection, createConnection } from 'typeorm';
+
+import rpcClient from '../components/rpc-client/rpc-client';
+import { BlockEntity } from '../entity/block.entity';
+import { TransactionEntity } from '../entity/transaction.entity';
+
+async function updateUnconfirmedBlocks(connection: Connection) {
+  console.log('start update block>>>>', new Date().getSeconds(), 's');
+
+  const transactionRepo = connection.getRepository(TransactionEntity);
+  const blockRepo = connection.getRepository(BlockEntity);
+  const { height } = await blockRepo
+    .createQueryBuilder('block')
+    .select(['height'])
+    .orderBy({
+      timestamp: 'DESC',
+    })
+    .limit(1)
+    .getRawOne();
+  const txs = await transactionRepo
+    .createQueryBuilder()
+    .select(['height', 'blockHash', 'id'])
+    .where(
+      `blockHash is null and (height is null or height < ${
+        Number(height) - 1
+      })`,
+    )
+    .getRawMany();
+  if (txs.length) {
+    for (let i = 0; i < txs.length; i += 1) {
```

Comment:
NB: `.forEach` or `for (const tx of txs)` are preferrable

Reply:
I think, it's better in terms of performance
https://medium.com/tech-tajawal/loops-performances-in-node-js-9fbccf2d6aa6

Reply:
Agree, makes sense with large amount of data